### PR TITLE
Add ReStyles - a.k.a styled components

### PR DESCRIPTION
**Pull request by lancetipton**
> This is **part 2** of the Re-Theme updates
> Before reviewing, make sure [this PR](https://github.com/simpleviewinc/keg-hub/pull/115) is merged first

## Context

* I had an idea for a way to do styles a little differently, but still common in the React world
* This pattern follows the `styles` components pattern, only it uses Re-Theme and the style-injector
  * This means it works for both web / native platforms

## Goal

* Add an additional way to style components

## Updates

* Removed the deprecated `theme.join` method
  * Updated the Demo app in a number of places where this was being used
  * But really, the Demo app will be removed, as soon as storybook is merged ( Part 3 of these PRs ) 
* Added a `reStyle` code export
  * This is a separate export, so it does not effect bundle size, unless it's imported
  * Allows styling components similar to how `styled` components works  
  * Added tests for reStyle hooks and export 

## Testing

**Test 1**
* Pull the PR => `keg pr 116`
* Run tests for jsutils => `keg jsutils && yarn test`
* Run tests for ReTheme => `keg retheme && yarn test`

**Test 2**
* Run this `Re-theme` docker package => `keg retheme pack run docker.pkg.github.com/simpleviewinc/keg-packages/retheme:add-storybook-to-retheme`
  * In your browser navigate to http://retheme-add-storybook-to-retheme.local.kegdev.xyz
  * Ensure everything works as expected

**Test 3**
* Run this `Keg-Components` docker package => `keg dpg run docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:add-storybook-to-retheme`
  * This has the `jsutils` and  `ReTheme` changes synced into it
*  In your browser navigate to http://components-add-storybook-to-retheme.local.kegdev.xyz/
  * Ensure **ALL** components render as expected

### CODE REVIEW

**Review by daniel112 — APPROVED** [review]
looks good

**Review by mpcarolin — APPROVED** [review]
I left a couple of comments/suggestions, but nothing blocking.

Overall looks good. Excited to use this 👍🏼

File: repos/re-theme/src/reStyle/reStyleHooks.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ * @param {string} styleProp - `style` || Custom style prop key to use
+ * @param {Object} props - Props of the component being wrapped
+ *
+ * @returns {Object} - Styles from the props
+ */
+export const useStyleProp = (styleProp, props) => {
+  return useMemo(() => {
+    return props[styleProp] || noOpObj
+  }, [ styleProp, props[styleProp] ])
+}
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply repos/re-theme/src/reStyle/reStyleHooks.js:77]: It's memoizing the value at `props[styleProp]`.
 Basically memoizing the styles object passed to the component relative to the `styleProp`, this way we only cause re-render if the `styles` or `styleProp` have changed.

File: repos/re-theme/src/reStyle/reStyleHooks.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ * Hook to ensure the className prop is an array with the compName argument added to it
+ * @function
+ * @param {String|Array} className - Current class names already set
+ * @param {string} compName - Name of a component to be added as a class name
+ *
+ * @returns {Array} - Built array of class names with the compName name added to it
+ */
+export const usePropClassName = (className, compName) => {
+  return useMemo(() => {
+    const classArr = className ? eitherArr(className, [className]) : []
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply repos/re-theme/src/reStyle/reStyleHooks.js:41]: Ohh, nice catch. I'll add that update to the next [PR](https://github.com/simpleviewinc/keg-hub/pull/117)

File: repos/re-theme/src/reStyle/reStyleHooks.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mpcarolin) [reply repos/re-theme/src/reStyle/reStyleHooks.js:77]:
```js
```

[the remark this answers is not in the document]
Comment (lancetipton) [reply repos/re-theme/src/reStyle/reStyleHooks.js:77]: After looking at this further, the `useStyleProp` is not really doing much. So I'm removing it in this [PR](https://github.com/simpleviewinc/keg-hub/pull/117)

**Pull request merged**